src/bin/edit/main.rs

```diff
@@ -233,37 +239,48 @@ fn run() -> apperr::Result<()> {
 
 // Returns true if the application should exit early.
 fn handle_args(state: &mut State) -> apperr::Result<bool> {
+    let scratch = scratch_arena(None);
+    let mut paths: Vec<PathBuf, &Arena> = Vec::new_in(&*scratch);
     let mut cwd = env::current_dir()?;
-    let mut path = None;
 
     // The best CLI argument parser in the world.
-    if let Some(arg) = env::args_os().nth(1) {
+    for arg in env::args_os().skip(1) {
         if arg == "-h" || arg == "--help" || (cfg!(windows) && arg == "/?") {
             print_help();
             return Ok(true);
         } else if arg == "-v" || arg == "--version" {
             print_version();
             return Ok(true);
         } else if arg == "-" {
-            // We'll check for a redirected stdin no matter what, so we can just ignore "-".
-        } else {
-            let p = cwd.join(Path::new(&arg));
-            let p = path::normalize(&p);
-            if let Some(parent) = p.parent() {
-                cwd = parent.to_path_buf();
-            }
-            path = Some(p);
+            paths.clear();
+            break;
+        }
+        let p = cwd.join(Path::new(&arg));
+        let p = path::normalize(&p);
+        if !p.is_dir() {
+            paths.push(p);
         }
     }
 
+    if paths.len() > 1 {
+        for p in &paths {
+            state.documents.add_file_path(p)?;
+        }
+    } else if paths.len() == 1 {
+        let p = &paths[0];
+        if let Some(parent) = p.parent() {
+            cwd = parent.to_path_buf();
+        }
+        state.documents.add_file_path(p)?;
+    }
```

Comment:
Could you separate the modification of `cwd` and the loop for `add_file_path` into two pieces? Something like this:
```suggestion
    for p in &paths {
        state.documents.add_file_path(p)?;
    }

    if let Some(parent) = paths.first().and_then(|p| p.parent()) {
        cwd = parent.to_path_buf();
    }
```

I think it's shorter and a bit easier to read.